### ISSUE TITLE
use of unstable library feature 'is_some_and'

### ISSUE THREAD

**Issue by Matthew494965** [Siriusmart/youtube-tui#44]
I'm getting this error message upon installation:

   --> /home/matthew/.cargo/registry/src/github.com-1ecc6299db9ec823/youtube-tui-0.8.0/src/global/structs/mpv.rs:113:14
    |
113 |             .is_some_and(|s| s.as_str() == "no" || s.as_str() == "false")
    |              ^^^^^^^^^^^
    |
    = note: see issue #93050 <https://github.com/rust-lang/rust/issues/93050> for more information

For more information about this error, try `rustc --explain E0658`.
error: could not compile `youtube-tui` due to previous error
warning: build failed, waiting for other jobs to finish...
error: failed to compile `youtube-tui v0.8.0`, intermediate artifacts can be found at `/tmp/cargo-installqIc8XJ`

**Comment by Siriusmart** [comment]
The project uses the stable rust compiler i believe, try updating rust with `rustup update`

**Comment by Siriusmart** [comment]
hi, is this resolved?

**Comment by Matthew494965** [comment]
Unfortunately not. For some reason, it is still not working. I will try contacting Google support again to see if they can resolve this.

--
Matt
________________________________

**Comment by Siriusmart** [comment]
Ok first of all, this is a third party interface for YouTube, and does not use Google services in any way. So there is no reason to contact Google support for this project.

Also, can you confirm what is your rustc version by typing `rustc -V`? The rust compiler on my system is `rustc 1.74.0 (79e9716c9 2023-11-13)`, maybe you are running an older version of rust.

If that doesn't work as well, you might want to install the nightly compiler with rustup and create a `rust-toolchain.toml` at root and paste in the following content to specify the nightly compiler (I don't think you'll get to this step, updating the compiler should work).

```toml
[toolchain]
channel = "nightly"
```

**Comment by Siriusmart** [comment]
As discussed in [this issue](https://github.com/rust-lang/rust/issues/93050), it seems like your rust compiler is below version 1.70, as this issue is resolved after 1.70.0.

**Comment by Matthew494965** [comment]
I apologize. I got you confused with a different developer who also created
a YouTube interface program. It was his program that required me to contact
Google support in order to resolve the issue (the issue with his program
was a problem I had in authenticating his account with youtube.

The rust compiler that I was using was downloaded directly from the
repository of the Linux mint version I was using. I had no way of knowing
whether my version of the rust compiler was up to date or not.

Finally, there is absolutely no need to get snippy with me. To be blunt, I
have had so many problems with Linux in general I decided to switch back to
Windows 11. I have not had problems with Windows 11 since switching back.
